### PR TITLE
Promote some JSON behavior to FieldBinding

### PR DESCRIPTION
**Pull request by swankjesse**
This way it's easier to share between JSON frameworks.

### CODE REVIEW

File: wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/StructTest.kt [hunk 1]
```diff
@@ -416,7 +414,6 @@ class StructTest {
       |}""".trimMargin()
     // Protoc doesn't print those unless explicitly set.
     val moshiJson = """{
-      |  "nullValue": null,
```
Comment (swankjesse) [inline wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/StructTest.kt:419]: I didn’t set out to make these deltas; they came out accidentally. Any concerns? I think probably the thing we want to be most deliberate about is when to omit null. (Always?)

File: wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/FieldBinding.kt [hunk 2]
```diff
@@ -142,4 +181,83 @@ class FieldBinding<M : Message<M, B>, B : Message.Builder<M, B>> internal constr
...
 
   internal fun getFromBuilder(builder: B): Any? = builderField.get(builder)
+
+  fun jsonStringAdapter(syntax: Syntax): JsonFormatter<*>? {
+    return when (adapterString) {
+      "com.squareup.wire.ProtoAdapter#BYTES" -> ByteStringJsonFormatter
+      "com.squareup.wire.ProtoAdapter#INT64",
+      "com.squareup.wire.ProtoAdapter#SFIXED64",
+      "com.squareup.wire.ProtoAdapter#SINT64" -> LongAsStringJsonFormatter
+      "com.squareup.wire.ProtoAdapter#FIXED64" -> UnsignedLongAsStringJsonFormatter
```
Comment (oldergod) [inline wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/FieldBinding.kt:191]: Do we wanna use them for proto2 as well? I was thinking we'd add that for proto3 only.

File: wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/StructTest.kt (hunk 1 above)
Comment (oldergod) [reply wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/StructTest.kt:419]: All good

File: wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/FieldBinding.kt (hunk 2 above)
Comment (swankjesse) [reply wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/FieldBinding.kt:191]: Good call.

Comment (oldergod) [reply wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/FieldBinding.kt]: That's was probably why the tests were failing. It's green now!

**Pull request merged**